File: Data Structures & Algorithms/can-place-flowers/submission-3.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        nums = [0] + flowerbed + [0]
        i = 1
        while (i < len(nums) - 1):
            if (nums[i] == 0 and nums[i - 1] == 0 and nums[i + 1] == 0):
                n -= 1
                i += 1
            i += 1

        return n <= 0

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-3.java
class Solution {
    public int lengthOfLIS(int[] nums) {
        int n = nums.length;
        List<Integer> temp = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            if (temp.isEmpty()) {
                temp.add(nums[i]);
            } else {
                if (nums[i] > temp.getLast()) {
                    temp.add(nums[i]);
                } else {
                    int idx = Collections.binarySearch(temp, nums[i]);
                    if (idx >= 0)
                        continue;
                    idx = -(idx) - 1;
                    temp.set(idx, nums[i]);
                }
            }
        }
        return temp.size();
    }
}
